subsys/net/ip/net_shell.c

```diff
@@ -3047,16 +3051,22 @@ static enum net_verdict handle_ipv4_echo_reply(struct net_pkt *pkt,
 static int ping_ipv4(const struct shell *shell,
 		     char *host,
 		     unsigned int count,
-		     unsigned int interval)
+		     unsigned int interval,
+		     int iface_idx)
 {
 	struct in_addr ipv4_target;
+	struct net_if *iface;
```

Comment:
Similar changes to ipv4, so here `struct net_if *iface = net_if_get_by_index(iface_idx);`